openraft/src/core/admin.rs

```diff
@@ -229,8 +285,11 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
 
         let membership = &self.core.effective_membership.membership;
 
+        // remove nodes which not included in nodes, nor learners
+        let all_nodes = membership.all_nodes();
+        let all_learners = membership.all_learners();
         for (id, state) in self.nodes.iter_mut() {
-            if membership.contains(id) {
+            if all_nodes.contains(id) || all_learners.contains(id) {
```

Comment:
What about letting `contains()` check if a member or learner presents in the `Membership`, and adding another two method `is_member(node_id)` and `is_learner(node_id)`?